flexget/plugins/input/imdb_watchlist.py

```diff
@@ -99,57 +115,109 @@ def on_task_input(self, task, config):
             raise plugin.PluginError(e.args[0])
         if page.status_code != 200:
             raise plugin.PluginError('Unable to get imdb list. Either list is private or does not exist.')
+        return page
+
+    def parse_react_widget(self, task, config, url, params, headers):
+        page = self.fetch_page(task, url, params, headers)
+#         log.debug(page.text)
+        re_result = re.search('IMDbReactInitialState.push\((.+?)\);\n', page.text)
+        json_text = re_result.group(1)
```

Comment:
This will crash if it doesn't find a match. Should be wrapped in a try-except and spit out a suitable error message informing the user that the plugin may be broken

Reply:
Thanks, cvium for the thorough code review!
I left line 121 out of the try-except, since the fetch_page function has its own code that handles HTTPErrors or wrong http status codes.